pkg/scheduler/framework/plugins/nodelabel/node_label.go

```diff
@@ -63,7 +52,7 @@ func validateNoConflict(presentLabels []string, absentLabels []string) error {
 
 // New initializes a new plugin and returns it.
 func New(plArgs *runtime.Unknown, handle framework.FrameworkHandle) (framework.Plugin, error) {
-	args := Args{}
+	args := schedulerv1alpha2.NodeLabelArgs{}
```

Comment:
Use internal version from `"k8s.io/kubernetes/pkg/scheduler/apis/config"`

Reply:
The internal versions do not have json tags. And the versioned API types can be used when creating the configuration files for the in-tree plugins.

If we use the internal types here, we'd need to test them as the plugin code would use them. And we would also have to keep a separate set of tests for the API types as somebody might use them to generate plugin configuration.

Please note that this will change once the decisions around conversions are made in https://github.com/kubernetes/kubernetes/issues/88093. And the long term goal is to use the internal types here.

Reply:
SG.